src/compiler/chainrules.jl

```diff
@@ -0,0 +1,127 @@
+const chainrules_fallback = which(rrule, Tuple{Any})
+
+"""
+  has_chain_rrule(T)
+
+For a type-tuple `T` e.g. `Tuple{typeof(f), Int, Float64}`, checks if there is a `rrule` defined for it.
+Excluding the generic fallback.
```

Comment:
maybe this is meant to be expanded

Reply:
how so?
Note this is an internal docstring, so it is for people who are areading the code along-side.